src/app/d2-loadout-builder/locked-armor/LockedItem.tsx

```diff
@@ -35,42 +36,50 @@ export default function LockedItem({
   }
 
   // Multi-things locked
-  if (locked.items.length > 1) {
+  if (locked.length > 1 || locked[0].type === 'exclude') {
+    const perks = locked.filter((item) => item.type === 'perk');
+    const excluded = locked.filter((item) => item.type === 'exclude');
+
     return (
       <div className="empty-item" onClick={toggleOpen}>
         <div
           className={classNames('add-perk-container', {
-            'has-locked': locked.items.length
+            'has-locked': locked.length
           })}
         >
           <div className="add-perk-text">
-            {locked.items.length} {locked.type === 'exclude' ? 'excluded' : 'locked perks'}
+            {perks.length !== 0 && (
+              <div>{t('LoadoutBuilder.LockedPerks', { locked: perks.length })}</div>
+            )}
+            {excluded.length !== 0 && (
+              <div>{t('LoadoutBuilder.ExcludedItems', { locked: excluded.length })}</div>
+            )}
           </div>
         </div>
       </div>
     );
   }
 
-  const item = locked.items[0];
+  const lockedItem = locked[0];
 
   // one item locked/excluded
-  if (locked.type === 'exclude' || locked.type === 'item') {
+  if (lockedItem.type === 'item') {
```

Comment:
You can add `&& lockedItem.item.isDestiny2()` and then `lockedItem.item` will be a `D2Item` within the branch.